src/lib.rs

```diff
@@ -678,6 +692,36 @@ mod fv_scalar_tests {
         assert_eq!(pt_after_add, vplusw);
     }
 
+    #[test]
+    fn test_sub_plain() {
+        let fv = FV::<Scalar>::default_2048();
+        let sk = fv.generate_key();
+
+        let mut v = vec![0; fv.n];
+        for i in 0..fv.n {
+            v[i] = i as u8;
+        }
+
+        let mut w: Vec<u8> = vec![];
+        for i in 0..fv.n {
+            w.push((fv.n + i) as u8);
```

Comment:
fv.n is 2048 which will overflow u8 already -- maybe use some smaller numbers?

Reply:
same as the addition tests though

Reply:
actually, that's right! Could you fix the addition tests as well? Really appreciate it!